app/models/permit_application.rb

```diff
@@ -55,6 +53,12 @@ def number_prefix
     jurisdiction.prefix
   end
 
+  def assign_default_nickname
+    nickname = "#{jurisdiction_name}: #{full_address || pid || pin || id}"
```

Comment:
i think we can simplify this to 1 line, the return should happen automaticlaly.

self.nickname= "#{jurisdiction_name}: #{full_address || pid || pin || id}"